src/pieces/commands/autocommit.py

```diff
@@ -0,0 +1,80 @@
+import subprocess
+import re
+from pieces.gui import show_error
+from pieces.api.config import pos_client,api_client
+import re
+
+def get_current_working_changes(word_limit:int=2000) -> str:
+    """
+    Fetches the detailed changes in the files you are currently working on, limited to a specific word count.
+    
+    Args:
+        word_limit (int): The maximum number of words to include in the summary.
+    
+    Returns:
+        A string summarizing the detailed changes in a format suitable for generating commit messages,
+        truncated to the specified word limit.
+    """
+    
+    result = subprocess.run(["git", "diff"], capture_output=True, text=True)
+    detailed_diff = result.stdout.strip()
+    
+    summary = ""
+    for line in detailed_diff.split('\n'):
+        if line.startswith('diff --git'):
+            file_changed = re.search(r'diff --git a/(.+) b/\1', line)
+            if file_changed.group(1).endswith("poetry.lock"):
+                continue
+            if file_changed:
+                summary += f"File changed: **{file_changed.group(1)}**\n"
+        elif line.startswith('+') and not line.startswith('+++'):
+            summary += "Addition: " + line[1:].strip() + "\n"
+        elif line.startswith('-') and not line.startswith('---'):
+            summary += "Deletion: " + line[1:].strip() + "\n"
+    
+    
+    # Truncate the summary to the specified word limit
+    truncated_summary = summary[:word_limit-340] # removeing 340 characters because of the prompt characters
+    return truncated_summary
+
+
+def git_commit(**kwargs):
```

Comment:
would be solid to get a description to this function as well, any sort of limitations, how this should be expected to work